core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -150,4 +150,30 @@ void shouldRecognizeInvalidGitHubUrl() throws Exception {
         assertThat(r1.status()).isEqualTo(ResultStatus.FAILURE);
         assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid");
     }
+
+    @Test
+    void shouldReturnCorrectScmFolderPath() throws IOException {
+        final Plugin p1 = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub github = mock(GitHub.class);
+        final String repositoryName = "jenkinsci/test-repo";
+
+        when(p1.getScm()).thenReturn("https://github.com/" + repositoryName);
+        when(p1.getDetails()).thenReturn(Map.of(
+            UpdateCenterPluginPublicationProbe.KEY, ProbeResult.success(UpdateCenterPluginPublicationProbe.KEY, "")
+        ));
+        when(ctx.getGitHub()).thenReturn(github);
+        GHRepository repository = ctx.getGitHub().getRepository(repositoryName);
+        when(github.getRepository(repositoryName)).thenReturn(repository);
+
+        when(p1.getName()).thenReturn("test-repo");
+
+        final SCMLinkValidationProbe probe = getSpy();
+        final ProbeResult r1 = probe.apply(p1, ctx);
+
+        assertThat(ctx.getScmFolderPath()).isEqualTo("test-repo");
```

Comment:
this is incorrect. From what I see, the `scmFolderPath` should be `test-nested-dir-1`

```suggestion
        assertThat(ctx.getScmFolderPath()).isEqualTo("test-nested-dir-2");
```